### ISSUE TITLE
RFC: Including the Typing returns in Metrics

### ISSUE THREAD

**Issue by mkamioner** [awslabs/aws-lambda-powertools-python#642]
## Key information

* RFC PR: (leave this empty)
* Related issue(s), if known:
* Area: Metrics
* Meet [tenets](https://awslabs.github.io/aws-lambda-powertools-python/#tenets): (Yes/no)
* Approved by: ''
* Reviewed by: ''

## Summary
[summary]: #summary

The Metrics library does not include the return types of the functions, causing MyPy to complain.

## Motivation
[motivation]: #motivation

Adding typing to the return of the signature of the function will help prevent errors during development. Also, Mypy will stop complaining:

## Proposal
[proposal]: #proposal

Go through the metrics/metrics.py file and add the return type to all functions

## User Experience

This would require no code change and be backwards compatible. Users would get a better result from mypy and type hinting tools to help makde development easier

**Before**

```sh
/file.py: note: In member "record_metrics" of class "ConnectionCountRecorderDriver":
/file.py:26:9: error: Call to untyped function "clear_metrics" in typed context  [no-untyped-call]
Found 1 errors in 1 file (checked 1 source file)
```

**After**

```sh
Success: no issues found in 1 source file
```

## Drawbacks
[drawbacks]: #drawbacks

Honestly, none that I can think of except for someone's time - happy to take care of that :-)

## Rationale and alternatives
[rationale-and-alternatives]: #rationale-and-alternatives

* **What other designs have been considered? Why not them?** Not sure what else has been considered, but since we use python typing in other places in the same file, it makes sense to be consistent.
* **What is the impact of not doing this?** Better developer experience 💪 

## Unresolved questions
[unresolved-questions]: #unresolved-questions

> Optional, stash area for topics that need further development e.g. TBD

**Comment by heitorlessa** [comment]
Hey @mkamioner, we absolutely agree and if you could help us make MyPy compliant we'd appreciate it immensely!

We've fixed roughly 600 of the initial MyPy findings[1].... there's still 35+ from my last count to be fixed (Metrics included).

[1] https://github.com/awslabs/aws-lambda-powertools-roadmap/issues/3

**Comment by heitorlessa** [comment]
hey @mkamioner - While I couldn't reproduce that particular error with Metrics (a snippet would be great), I've gone ahead and added return types `-> None, really` to all of them, and fixed other pending areas.

In the overall project we've now got 13 mypy issues to fix but that should in theory address your problem in the next release.

Thanks!

**Comment by heitorlessa** [comment]
This is now available as part of 1.21.0, do let us know otherwise (and if you do, please share a snippet and mypy config to help us reproduce)

https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v1.21.0
